Extended messages cannot have required fields (must be optional).

Quix0r opened the pull request:
There is an error when I use the proto file _db_query_record_msg.proto_:

_db_query_record_msg.proto:36:11: Message extensions cannot have required fields.
/home/quix0r/git/seeks/src/plugins/query_capture/Makefile:850: recipe for target 'db_query_record_msg.pb.cc' failed
make: **\* [db_query_record_msg.pb.cc] Error 1_

This small change fixes it.

Quix0r committed:
Extended messages cannot have required fields (must be optional).

Signed-off-by: Roland Haeder <roland@mxchange.org>

beniz commented:
Thanks. this might be due to using a new protocol buffer version. What version are you using ?

Quix0r commented:
_$ protoc --version
libprotoc 2.6.0
$_

Is my change working with older versions? Then, I think, it is save to merge. :-)

Quix0r committed:
Please use full PHP header <?php and not the (deprecated) <? short varriant.

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
Heavy improvements:
- header('Location: bla-blub'); does *NOT* abort the script, *ALWAYS* use an explicit exit() call!
- Check for $_SERVER array elements, not all browsers may send it causing E_NOTICE (bad code style)
- Converted double-quotes to single-quotes
- if() and else conditions are now in one line + curly braces are always explicitly set
- Other small cleanups

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
More cleanups

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
'Seeks-Remote-Location' is a custom HTTP header and according to specification
it *MUST* be prefixed with 'X-'.

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
Temporary hide-out ... ;-)

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
No, better let Beniz change this ... ;-)

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
Better check these arrays.

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
Is it HOME_PAGE_URL that needs to be changed?

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
Some changes

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
Added missing template

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
Attempt to identify this one:

Error: 1 (Out of memory) in plugin websearch-api-compat caught in top-level handler

This only happens on my LAN instance, not the one at https://seeks.mxchange.org/search.php ...

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
Add 'forbidden' template here as well.

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
Let also solve errors from udb_service in errlog::sp_err_to_string()

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
Also need to add include search path here.

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
Use URL-encoding for binary data to safe-guard it + a lot (!) debug lines added.

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r committed:
Some cosmetics applied

Signed-off-by: Roland Haeder <roland@mxchange.org>

Quix0r commented:
:-( All commits end up here in one merge request ...

```
Add more commits by pushing to the master branch on Quix0r/seeks.
```

They should have made this notice bigger.

beniz commented:
Thanks, I'll take time to review this.

Quix0r committed:
"X-" has been deprecated:
http://tools.ietf.org/html/rfc6648

Signed-off-by: Roland Haeder <roland@mxchange.org>

beniz commented:
My apologies I was taken elsewhere and missed this PR. Is this still relevant ? Are you using Seeks ?

GreenLunar commented:
I am using Seeks every day.  I am willing to test whatever there is to test in this PR.
